src/preamble.js

```diff
@@ -1250,4 +1250,8 @@ function runMemoryInitializer() {
 var asyncifyStubs = {};
 #endif
 
+#if EMBIND_AOT_JS
+// <<< EMBIND_AOT_JS_OUTPUT >>>
+#endif
```

Comment:
Can we add this as a `--post-js` instead of modifying the preamble?

Reply:
I don't love this part of the change.. I hope we can fine a way to do this without putting this directly into the preamble like this. 

Perhaps we can simple append the code to the generated JS rather than replacing something deep inside it?

Reply:
I see now, I thought you meant add the generated code using post js, but you're saying add the `<<< EMBIND_AOT_JS_OUTPUT >>>` in a post js file and still do the replacement.

Reply:
Actually I was hoping to avoid the replacement.. I guess I don't understand why you can't put the contents of EMBIND_AOT_JS_OUTPUT into a POST_JS_FILES element?

Reply:
Oh wait I see now .. the EMBIND_AOT_JS_OUTPUT is not known until after the first program runs.

Reply:
> I was doing this originally, but the embind tests were failing. Currently the embind test's pre and post js wrap the module, which caused the generated methods to then be outside of the closure. Do you know if users do something like this too?
> 
> e.g. test.pre.js
> 
> ```js
> (function(){return this;})()['Module'] = (function() {
> ```
> 
> test.post.js
> 
> ```
>   return Module;
> })();
> ```

What about if you make the first lement of post_js list.. so it comes before the `return Module;`?

Reply:
I still don't think that will work because of the current ordering:

1. build the program normally (this generates the final output js which already has POST/PRE js)
2. build the program with AOT generation code
3. run the above in node (this generates EMBIND_AOT_JS_OUTPUT)
4. inject EMBIND_AOT_JS_OUTPUT back into the output from step 1

Reply:
We could try to do the AOT generation first, and then put that in a post js file. However, then I'll run into the build performance issue that I ran into with typescript generation.

Reply:
But don't you already run the AOT generation with and empty POST_JS_FILES?  i.e. post JS files should not effect the AOT process?

What are the build performance issues?

Reply:
Ok, lets leave this as it is for now then.

But can you maybe remove the `//` here (and the `\n` in the replacement) by using a new internal settings such as `EMBIND_INCLUDE_AOT_OUTPUT`   which wraps this?